src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs

```diff
@@ -778,13 +783,56 @@ public static async Task<IEnumerable<IStorageItemWithPath>> GetDraggedStorageIte
 					return itemsList;
 				}
 			}
-			if (packageView.Properties.TryGetValue("FileDrop", out var data))
+
+			// workaround for GetStorageItemsAsync() bug that only yields 16 items at most
+			// https://learn.microsoft.com/en-us/windows/win32/shell/clipboard#cf_hdrop
+			if (packageView.Contains("FileDrop"))
 			{
-				if (data is List<IStorageItemWithPath> source)
+				var fileDropData = await packageView.GetDataAsync("FileDrop");
+				if (fileDropData is IRandomAccessStream stream)
 				{
-					itemsList.AddRange(source);
+					stream.Seek(0);
+
+					byte[] dropBytes = new byte[stream.Size];
+					int bytesRead = await stream.AsStreamForRead().ReadAsync(dropBytes);
+
+					if (bytesRead > 0)
+					{
+						IntPtr dropStructPointer = Marshal.AllocHGlobal(dropBytes.Length);
+
+						try
+						{
+							Marshal.Copy(dropBytes, 0, dropStructPointer, dropBytes.Length);
+							HDROP dropStructHandle = new(dropStructPointer);
+
+							var itemPaths = new List<string>();
+							uint filesCount = Shell32.DragQueryFile(dropStructHandle, 0xffffffff, null, 0);
+							for (uint i = 0; i < filesCount; i++)
+							{
+								string buffer = new('\0', Kernel32.MAX_PATH);
+								uint charsCopied = Shell32.DragQueryFile(dropStructHandle, i, buffer, Kernel32.MAX_PATH);
```

Comment:
I'd like to avoid using Kernel32.MAX_PATH. Perhaps something like:
```cs
uint charsNeeded = Shell32.DragQueryFile(dropStructHandle, i, NULL 0);
string buffer = new('\0', charsNeeded);
uint charsCopied = Shell32.DragQueryFile(dropStructHandle, i, buffer, charsNeeded);
```

Reply:
i'm onboard with that, it's non-invasive and has benefits of saving memory and avoiding platform issues